library/src/main/java/com/getstream/sdk/chat/rest/adapter/MessageGsonAdapter.java

```diff
@@ -24,7 +24,13 @@
 
     @Override
     public void write(JsonWriter writer, Message message) throws IOException {
+
         HashMap<String, Object> data = new HashMap<>();
+        TypeAdapter adapter = GsonConverter.Gson().getAdapter(HashMap.class);
+        if (message == null){
```

Comment:
Still weird and looks like a workaround. 
Lets leave a comment at least mentioning why there is this check.

Also googled briefly and apparently there was a bug before Gson 2.4: https://stackoverflow.com/a/55884142/798165
But we use the latest version. The link also I think should be mentioned in the comment.